gwdatafind/utils.py

```diff
@@ -137,6 +140,66 @@ def find_credential():
                        "Please run 'grid-proxy-init -rfc' and try again.")
 
 
+def find_scitoken(aud, scope):
+    """Locate a Scitoken for a given audience and scope.
+
+    This function checks for a SciToken using the WLCG token discovery
+    protocol. If token is found that matches the requested audience and scope
+    then it is returned. If no matching token is found using this protocol,
+    the function checks to see if HTCondor has created a SciToken with the
+    appropriate audience and scope in the _CONDOR_CREDS directory. Tokens in
+    _CONDOR_CREDS that do not have the correct audience and scope are ignored.
+
+    Parameters
+    ----------
+    aud : `str`
+        the audience of the scitoken
+
+    scope : `str`
+        the scope of the scitoken
+
+    Returns
+    -------
+    token_data : `SciToken`
+        the located token or None if no token for the specified
+        audience can be found
+
+    """
+    scheme, path = scope.split(':')
+    token = None
+
+    # look for a token in with WLCG Bearer Token Discovery protocol
+    try:
+        token = scitokens.SciToken.discover(audience=aud)
+        token_enforcer = scitokens.Enforcer(token['iss'], audience=aud)
+        if token_enforcer.test(token, scheme, path) is False:
+            raise ValueError
+    except:
```

Comment:
Please don't use an empty `except` clause, since we probably don't want to catch things like `TypeError`s.